app/modules/sightings/resources.py

```diff
@@ -57,21 +80,182 @@ def get(self, args):
+
+        response = current_app.edm.request_passthrough(
+            'sighting.data', 'post', {'data': data}, ''
+        )
+
+        response_data = None
+        result_data = None
+        if response.ok:
+            response_data = response.json()
+            result_data = response_data.get('result', None)
+
+        if (
+            not response.ok
+            or not response_data.get('success', False)
+            or result_data is None
+        ):
+            log.warning('Sighting.post failed')
+            passed_message = {'message': {'key': 'error'}}
+            if response_data is not None and 'message' in response_data:
+                passed_message = response_data['message']
+            abort(success=False, passed_message=passed_message, message='Error', code=400)
+
+        # if we get here, edm has made the sighting.  now we have to consider encounters contained within,
+        # and make houston for the sighting + encounters
+
+        # FIXME must cleanup edm cruft!
+        if ('encounters' in data and 'encounters' not in result_data) or (
+            'encounters' not in data and 'encounters' in result_data
+        ):
+            log.error(
+                'Sighting.post missing encounters in one of %r or %r'
+                % (data, result_data)
+            )
+            _cleanup_post_and_abort(
+                None, None, 'Missing encounters between data and result'
+            )
+        if not len(data['encounters']) == len(result_data['encounters']):
+            log.error(
+                'Sighting.post imbalanced encounters in %r or %r' % (data, result_data)
+            )
+            _cleanup_post_and_abort(
+                None, None, 'Imbalance in encounters between data and result'
+            )
+        # i think this was official declared as law today 2021-02-24
+        if len(data['encounters']) < 1:
+            log.error('Sighting.post empty encounters in %r / %r' % (data, result_data))
+            _cleanup_post_and_abort(None, None, 'Must have at least one encounter')
+
+        # now we handle asset-related json that came in. note: arrays should be parallel in data/result_data
+        # this makes sure assetReferences is well-formed and also maps (future) assets to encounters
+
+        # NOTE for simplicity i am going to assume assetReferences *share a common transaction id*  !!
+        #  this will very likely always be true, but in the event it proves not to be, this will have to be altered
+        #  to do multiple submission.import_tus_files() calls below
+
+        arefs_found = {}
+        i = 0
+        while i < len(data['encounters']):
+            enc_data = data['encounters'][i]
+            # TODO handle regular .assets flavor
+            # TODO genericize this across here and encounters
+            if 'assetReferences' in enc_data and isinstance(
+                enc_data['assetReferences'], list
+            ):
+                enc_data['_paths_wanted'] = []
```

Comment:
This is modifying an input parameter... I think which is then used in _enc_assets.
I think it would be clearer if this was a separate hash to make it abundantly clear that data (request_in) goes into this block and arefs_found and paths_wanted comes out.

Reply:
yup, now switched to `paths_wanted[i]` for the _i-th_ encounter.  doing more work on this stuff today tho; so code in flux.